### PR TITLE
Updated release documentation

### PR DESCRIPTION
**Pull request by gsidebo**
Moved around some info to make this a more helpful guide for the actual release process

### CODE REVIEW

**Comment by gsidebo** [comment]
@annagav @pdpinch FYI. this should be an easy-enough review

File: README.md [hunk 1]
```diff
@@ -4,26 +4,30 @@
...
+    *(NOTE: This can be done using the 'Merge Pull Request' button in the Github PR issue. Do not 
+    delete the branch after merging)*
+1. Tag the ``release`` branch with the version number and push the tag to
+    the remote. For example, these commands create an annotated tag for
+    version 0.3.0 and push the tag to the remote. (You don't need to specify
+    the sha of the commit if you're on the branch you want to tag.)
+    ```
+    git tag -a -m "Release 0.3.0" v0.3.0 <sha of commit>
+    git push --follow-tags
+    ```
```
Comment (pdpinch) [inline README.md]: For some reason this block is rendering inline. I think maybe it needs a newline between lines 19 and 20.

File: README.md [hunk 2]
```diff
@@ -4,26 +4,30 @@
...
-10. Merge the ``release`` branch into the ``master`` branch,
-    and push ``master`` to ``origin`` (manual step)
-11. Send email notifications (manual step)
+A full release can be performed by following the steps below:
+
+1. Run release.sh to create a PR for the release (see details in the 
+    ["How to use release.sh"](#how-to-use-releasesh) section below)
+1. Inform the team that a release PR is up and that they need to verify their commits
+1. Once developers verify their commits, merge the 'release-candidate' branch into 'release', and push
+    the release branch.
```
Comment (pdpinch) [inline README.md]: Sometimes we use single quotes and sometimes we use backticks around the branch names. We should be consistent. 

I like backticks, I guess.

**Comment by gsidebo** [comment]
comments addressed

**Pull request merged**